### PR TITLE
Issue#1 Compile apps/memory-server

### PR DESCRIPTION
**Pull request by Arsnael**
PR on the work already done by chibenwa to have a proper review

### CODE REVIEW

File: openpaas-james/pom.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        <artifactId>openpaas-james-parent</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>openpaas-james</artifactId>
+    <packaging>pom</packaging>
+
+    <name>OpenPaaS James</name>
+    <description>Apache James extensions for OpemPaaS</description>
```
Comment (LanKhuat) [inline openpaas-james/pom.xml:17]: s/OpemPaaS/OpenPaaS

File: openpaas-james/apps/memory/src/test/java/com/linagora/openpaas/james/app/MemoryServerTest.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.linagora.openpaas.james.app;
+
+import org.apache.james.JamesServerBuilder;
+import org.apache.james.JamesServerContract;
+import org.apache.james.JamesServerExtension;
+import org.apache.james.jmap.draft.JmapJamesServerContract;
+import org.apache.james.modules.TestJMAPServerModule;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class MemoryServerTest  implements JamesServerContract, JmapJamesServerContract {
```
Comment (LanKhuat) [inline openpaas-james/apps/memory/src/test/java/com/linagora/openpaas/james/app/MemoryServerTest.java:10]: extra space

File: openpaas-james/apps/memory/src/test/resources/imapserver.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+
```
Comment (LanKhuat) [inline openpaas-james/apps/memory/src/test/resources/imapserver.xml:22]: extra line

File: openpaas-james/apps/memory/src/test/resources/pop3server.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                                                               
+  Unless required by applicable law or agreed to in writing,   
+  software distributed under the License is distributed on an  
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       
+  KIND, either express or implied.  See the License for the    
+  specific language governing permissions and limitations      
+  under the License.                                           
+ -->
+
+
```
Comment (LanKhuat) [inline openpaas-james/apps/memory/src/test/resources/pop3server.xml:21]: extra line

**Review by LanKhuat — APPROVED** [review]

**Comment by Arsnael** [comment]
I approve this PR

**Review by remk — APPROVED** [review]

**Comment by Arsnael** [comment]
Merged

File: openpaas-james/jmap/pom.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rouazana) [inline openpaas-james/jmap/pom.xml:1]: missing license?

File: openpaas-james/jmap/pom.xml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (rouazana) [inline openpaas-james/jmap/pom.xml:2]: missing license?

File: openpaas-james/jmap/src/main/java/com/linagora/openpaas/james/jmap/FiltersGetMethod.java [hunk 7]
```diff
@@ -0,0 +1,2 @@
+package com.linagora.openpaas.james.jmap;public class FiltersGetMethod {
```
Comment (rouazana) [inline openpaas-james/jmap/src/main/java/com/linagora/openpaas/james/jmap/FiltersGetMethod.java]: missing license?

File: openpaas-james/pom.xml [hunk 8]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rouazana) [inline openpaas-james/pom.xml:1]: missing license?

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,308 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rouazana) [inline pom.xml:1]: missing license?

File: openpaas-james/jmap/src/main/scala/com/linagora/openpaas/james/jmap/CustomMethod.scala [hunk 10]
```diff
@@ -0,0 +1,27 @@
+package com.linagora.openpaas.james.jmap
```
Comment (rouazana) [inline openpaas-james/jmap/src/main/scala/com/linagora/openpaas/james/jmap/CustomMethod.scala:1]: missing license?

File: openpaas-james/apps/memory/pom.xml [hunk 11]
```diff
@@ -0,0 +1,100 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rouazana) [inline openpaas-james/apps/memory/pom.xml:1]: missing license?

File: openpaas-james/apps/memory/src/main/java/com/linagora/openpaas/james/app/MemoryServer.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
+package com.linagora.openpaas.james.app;
```
Comment (rouazana) [inline openpaas-james/apps/memory/src/main/java/com/linagora/openpaas/james/app/MemoryServer.java:1]: missing license?

File: openpaas-james/apps/memory/src/test/java/com/linagora/openpaas/james/app/MemoryServerTest.java [hunk 13]
```diff
@@ -0,0 +1,16 @@
+package com.linagora.openpaas.james.app;
```
Comment (rouazana) [inline openpaas-james/apps/memory/src/test/java/com/linagora/openpaas/james/app/MemoryServerTest.java:1]: missing license?

File: openpaas-james/apps/pom.xml [hunk 14]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rouazana) [inline openpaas-james/apps/pom.xml:1]: missing license?

**Comment by Arsnael** [comment]
@rouazana It's different for openpaas-james, that's not apache code. The license is declared at the root of the project, you don't need to copy/paste it in every file as I understood

**Comment by rouazana** [comment]
> @rouazana It's different for openpaas-james, that's not apache code. The license is declared at the root of the project, you don't need to copy/paste it in every file as I understood

In general I agree with this statement, but here as we prone to mix licenses I think it will be better to explicit the license in each file.

**Comment by Arsnael** [comment]
> In general I agree with this statement, but here as we prone to mix licenses I think it will be better to explicit the license in each file.

you mean that one https://github.com/linagora/openpaas-james/blob/master/LICENSE in each file? it's damn huge :o and I'm really not convinced it's necessary here

**Comment by rouazana** [comment]
> > In general I agree with this statement, but here as we prone to mix licenses I think it will be better to explicit the license in each file.
> 
> you mean that one https://github.com/linagora/openpaas-james/blob/master/LICENSE in each file? it's damn huge :o and I'm really not convinced it's necessary here

See https://wiki.linagora.com/doku.php?id=juridique:licence_des_produits_linagora last section "How to Apply These Terms"

**Comment by Arsnael** [comment]
Thanks for the pointer I had no idea regarding that. Will change that monday, or if someone wants to do it before...

**Pull request closed without merging**